### ISSUE TITLE
Typedoc is broken

### ISSUE THREAD

**Issue by mishaor** [TypeStrong/typedoc#785]
I can't compile my project's documentation.
Here's the log:
```
PS C:\Users\misha\Documents\Projects\Cutelog> npm run docs

> cutelog@1.0.0 docs C:\Users\misha\Documents\Projects\Cutelog
> typedoc --out ./docs --theme markdown --target es6 --module es2015 .

Loaded plugin typedoc-plugin-markdown

Using TypeScript 2.7.2 from C:\Users\misha\Documents\Projects\Cutelog\node_modules\typedoc\node_modules\typescript\lib
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1361)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1361)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1361)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1362)
 Declaration or statement expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1388)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1393)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1398)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1403)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1403)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1408)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1408)
 Declaration or statement expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1361)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1361)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1361)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1362)
 Declaration or statement expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1388)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1393)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1398)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1403)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1403)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1408)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1408)
 Declaration or statement expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es5.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1361)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1361)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1361)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1362)
 Declaration or statement expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1388)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1393)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1398)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1403)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1403)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1408)
 Expression expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1408)
 ';' expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1408)
 Declaration or statement expected.
Error: C:/Users/misha/Documents/Projects/Cutelog/node_modules/typescript/lib/lib.es6.d.ts(1408)
 ';' expected.
npm ERR! code ELIFECYCLE
npm ERR! errno 4
npm ERR! cutelog@1.0.0 docs: `typedoc --out ./docs --theme markdown --target es6 --module es2015 .`
npm ERR! Exit status 4
npm ERR!
npm ERR! Failed at the cutelog@1.0.0 docs script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\misha\AppData\Roaming\npm-cache\_logs\2018-06-03T17_53_40_745Z-debug.log
```
Node version: 8.11.2
Typescript version: 2.9.1

**Comment by Mikhus** [comment]
Yes, same problem, heads up for accepting PR with typescript version update

**Comment by dskvr** [comment]
Typedoc shouldn't prevent people from upgrading Typescript. Documentation reacts to development, it shouldn't dictate development. This is an anti-pattern.

**Comment by blink1073** [comment]
@dskvr, please bear in mind everyone working on this project is doing so out of good will.  The original project was written before public APIs were available in TypeScript to make it feasible for this library to work across versions.  Updating the project to use the new API is a substantial effort.  cf https://github.com/TypeStrong/typedoc/issues/378

**Comment by aciccarello** [comment]
Moving to a peer dependency is under consideration for a 1.0.0 release. Until then typedoc will continue to include typescript as a direct dependency. Since TypeDoc is currently released with TypeScript 3.1 I believe this issue should be resolved.

If you have an immediate issue, please see the [Updating guide](https://github.com/TypeStrong/typedoc/blob/2c23aac3cd11f8410f52908cc34cfe6d4d773ecd/UPDATING.md) for a workaround to TypeScript upgrades.
